backend/extra/intro-to-express/starter/src/routes/posts.js

```diff
@@ -1,7 +1,19 @@
 import express from "express";
-import logger from "../middleware/logger.js";
-import { getPosts } from "../controllers/postsController.js";
+import {logger} from "../middleware/logger.js";
+import {validationPost,
+  validationUpdate,} from "../middleware/validation.js";
+import {TimestampPost,TimestampUpdate} from "../middleware/timesatamp.js";
+
+import { getPosts,
+    getPostById,
+    createPost,
+    updatePost,
+    deletePost } from "../controllers/postsController.js";
 
 export const routes = express.Router();
 
 routes.get("/", logger, getPosts);
+routes.get("/:id",logger,getPostById);
+routes.post("/",[logger,validationPost,TimestampPost],createPost);
+routes.put("/:id",[logger,validationUpdate,TimestampUpdate],updatePost);
+routes.delete("/:id",logger,deletePost);
```

Comment:
Since logger is a global middleware, t9deri tkherjih 3amil mochtarak like this, hka t9oli el express beli all the requestion li yji ksh wahed men routes, lzm yjoz 9bel 3la logger :
```suggestion
// Apply logger middleware to all routes
routes.use(logger);

routes.get("/", getPosts);
routes.get("/:id", getPostById);
routes.post("/", [validationPost, TimestampPost], createPost);
routes.put("/:id", [validationUpdate, TimestampUpdate], updatePost);
routes.delete("/:id", deletePost);
```